packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -943,35 +973,57 @@ class AlignTransition extends AnimatedWidget {
   ///  * [Align.new].
   const AlignTransition({
     super.key,
-    required Animation<AlignmentGeometry> alignment,
-    required this.child,
+    required this.alignment,
     this.widthFactor,
     this.heightFactor,
-  }) : super(listenable: alignment);
+    required Widget super.child,
+  });
 
   /// The animation that controls the child's alignment.
-  Animation<AlignmentGeometry> get alignment => listenable as Animation<AlignmentGeometry>;
+  final ValueListenable<AlignmentGeometry> alignment;
 
   /// If non-null, the child's width factor, see [Align.widthFactor].
   final double? widthFactor;
 
   /// If non-null, the child's height factor, see [Align.heightFactor].
   final double? heightFactor;
 
-  /// The widget below this widget in the tree.
-  ///
-  /// {@macro flutter.widgets.ProxyWidget.child}
-  final Widget child;
-
   @override
-  Widget build(BuildContext context) {
-    return Align(
-      alignment: alignment.value,
+  RenderPositionedBox createRenderObject(BuildContext context) {
+    return _RenderAnimatedAlign(
+      listenable: alignment,
       widthFactor: widthFactor,
       heightFactor: heightFactor,
-      child: child,
+      textDirection: Directionality.maybeOf(context),
     );
   }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderPositionedBox renderObject) {
+    (renderObject as _RenderAnimatedAlign)
+      ..listenable = alignment
+      ..widthFactor = widthFactor
+      ..heightFactor = heightFactor
+      ..textDirection = Directionality.maybeOf(context);
+  }
+}
+
+class _RenderAnimatedAlign extends RenderPositionedBox with _RenderTransition<ValueListenable<AlignmentGeometry>> {
+  _RenderAnimatedAlign({
+    required ValueListenable<AlignmentGeometry> listenable,
+    super.widthFactor,
+    super.heightFactor,
+    super.textDirection,
+  }) : _listenable = listenable,
+       super(alignment: listenable.value);
+
+  @override
+  ValueListenable<AlignmentGeometry> _listenable;
+
+  @override
+  void _listener() {
+    alignment = _listenable.value;
```

Comment:
Maybe leave a comment here for future readers that setting the alignment to the new value will trigger a re-layout with that new value.